### ISSUE TITLE
The desision tree is different due to parameter random_state of function train_test_split

### ISSUE THREAD

**Issue by yltang52** [amueller/introduction_to_ml_with_python#114]
Hi there,

I have a question:

The following `decision tree` code is exactly the same as that in the text book (ch 2), except for the `random_state` parameter of the `train_test_split` function (I used 0 here instead of 42):

    from sklearn.model_selection import train_test_split
    from sklearn.tree import DecisionTreeClassifier
    from sklearn.datasets import load_breast_cancer
    from sklearn.tree import export_graphviz
    import graphviz

    cancer = load_breast_cancer()
    X_train, X_test, y_train, y_test = train_test_split(
        cancer.data, cancer.target, stratify=cancer.target, random_state=0)
    tree = DecisionTreeClassifier(random_state=0)
    tree.fit(X_train, y_train)
    print("Accuracy on training set: {:.3f}".format(tree.score(X_train, y_train)))
    print("Accuracy on test set: {:.3f}".format(tree.score(X_test, y_test)))

    tree = DecisionTreeClassifier(max_depth=4, random_state=0)
    tree.fit(X_train, y_train)

    print("Accuracy on training set: {:.3f}".format(tree.score(X_train, y_train)))
    print("Accuracy on test set: {:.3f}".format(tree.score(X_test, y_test)))

    export_graphviz(tree, out_file="tree.dot", class_names=["malignant", "benign"],
                    feature_names=cancer.feature_names, impurity=False, filled=True)

    with open("tree.dot") as f:
        dot_graph = f.read()
    display(graphviz.Source(dot_graph))

The resulting tree has the root node being `worst perimeter <= 106.1` instead of `worst radius <= 16.795`. Why is that? Thanks a lot.

**Comment by amueller** [comment]
If the data is different there is no reason for the tree to be the same. Trees are notoriously unstable as explained in the book.
